src/js/components/Products/Product.js

```diff
@@ -0,0 +1,191 @@
+        this.onSubmit=this.onSubmit.bind(this);
+        this.Small=this.Small.bind(this);
+        this.Medium=this.Medium.bind(this);
+        this.Large=this.Large.bind(this);
+        this.handleChecked=this.handleChecked.bind(this);
+        this.Color=this.Color.bind(this)
+    }
+    
+    onChange(e){
+        this.setState({[e.target.name]:e.target.value});
+    }
+
+    onSubmit(e){
+        e.preventDefault();
+        const post={
+            Name:this.state.Name,
+            Description: this.state.Description,
+            Price:this.state.Price,
+            Type:this.state.Type,
+            SizeOption:this.state.count,
+            ColourOption:this.state.ColourOption,
+            CoverImageLocation:this.state.FileUpload,
+        }
+
+        fetch(myConstClass.PRODUCT_URL,{
+            method:'POST',
+            headers:{
+                'Accept': 'application/json',
+                'content-type':'application/json'
+            },
+            body:JSON.stringify(post)
+        })
+        .then(response=>response.json());
+    }
+
+    Color()
+    {
+        if(this.state.color==false){
+            this.setState({color:true});
+            this.setState({ColourOption:1})
+        }
+        else{
+            this.setState({color:false})
+        }
+
+    }
+
+    handleChecked()
+    {
+        if(this.state.speedy==false){
+            this.setState({speedy:true,kyn:false,Type:"Speedy"});
+        }
+        else{
+            this.setState({speedy:false,kyn:true,Type:"Know your nation"});
+        }
+    }
+    
+    Small()
+    {
+        if(this.state.small==false)
+        {
+            this.setState({small:true});
+            this.setState({count:1}); 
+        }
+        else{
+            this.setState({small:false}); 
+        }
+    }
+
+    Medium()
+    {
+        if(this.state.medium==false){
+            this.state.medium=true;
+            this.setState({count:2})
+        }
+        else{
+            this.state.medium=false;
+        }
+    }
+
+    Large()
+    {
+        if(this.state.large==false){
+            this.setState({large:true});
+            this.setState({count:3});
+        }
+        else{
+            this.setState({large:false});
+        }
+    } 
+    
+    render()
+    {
+        let loc_navBarTitle = "CREATE PRODUCT";
+        let loc_navbarItems = [true, false, false, true];
+        return (
+            <div className="Product">
+                <Navbar titleFromParent={loc_navBarTitle} navbarItems={loc_navbarItems}/>
+                <div className="container ">
+                   <div>
+                        <img src={require('./download.jpg')} />
+                    </div>
+                    <div className="row ">
+                        <div className="col-12 cover-image">
+                            <label >Cover Image</label>
+                            <span>
+                                <input name="file" type="file"  onChange={this.handleChage}/> 
```

Comment:
this.handleChage if you spelled it correctly is not visible in your constructor.
Please add it there.

Reply:
I removed  handleChange function. now its sorted